File: backend/models/routeModels/user_settings.go
package routeModels

type UserSettings struct {
	HTTPProtocol     bool   `json:"http_protocol"`
	HTTPSProtocol    bool   `json:"https_protocol"`
	SOCKS4Protocol   bool   `json:"socks4_protocol"`
	SOCKS5Protocol   bool   `json:"socks5_protocol"`
	Timeout          uint16 `json:"timeout"`
	Retries          uint8  `json:"retries"`
	UseHttpsForSocks bool   `gorm:"not null;default:true"`
}

File: frontend/src/app/models/UserSettings.ts
export interface UserSettings {
  http_protocol:     boolean
  https_protocol:    boolean
  socks4_protocol:   boolean
  socks5_protocol:   boolean
  timeout:          number
  retries:          number
  UseHttpsForSocks: boolean
  auto_remove_failing_proxies: boolean
  auto_remove_failure_threshold: number

  judges: [{
    url: string
    regex: string
  }]

  scraping_sources: string[]
}
